src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -306,11 +343,35 @@ private static bool IsValidIssuer(string validIssuerTemplate, string tenantId, s
             }
         }
 
-        private BaseConfigurationManager GetEffectiveConfigurationManager(SecurityToken securityToken)
+        private static bool IsV2Issuer(SecurityToken securityToken)
         {
-            var isV2 = securityToken.Issuer.EndsWith(V2EndpointSuffixWithTrailingSlash, StringComparison.OrdinalIgnoreCase) ||
+            return securityToken.Issuer.EndsWith(V2EndpointSuffixWithTrailingSlash, StringComparison.OrdinalIgnoreCase) ||
                 securityToken.Issuer.EndsWith(V2EndpointSuffix, StringComparison.OrdinalIgnoreCase);
-            return isV2 ? ConfigurationManagerV2 : ConfigurationManagerV1;
+        }
+
+        private BaseConfigurationManager GetEffectiveConfigurationManager(SecurityToken securityToken)
+        {
+            var isV2Issuer = IsV2Issuer(securityToken);
+
+            if (_configurationManagerProvider != null)
+            {
+                var aadAuthority = isV2Issuer ? AadAuthorityV2 : AadAuthorityV1;
+                var configurationManager = _configurationManagerProvider(aadAuthority);
+
+                if (configurationManager != null)
+                    return configurationManager;
+            }
+
+            // If no provider or provider returned null, fallback to previous strategy
```

Comment:
This is where the fallback happens in case configurationManager fetched from provider was null. Is this a desired behavior?